src/ORM/DataList.php

```diff
@@ -308,72 +313,118 @@ public function distinct($value)
     }
 
     /**
-     * Return a new DataList instance as a copy of this data list with the sort
-     * order set.
+     * Return a new DataList instance as a copy of this data list with the sort order set
+     *
+     * Raw SQL is not accepted, only actual field names can be passed
      *
-     * @see SS_List::sort()
-     * @see SQLSelect::orderby
+     * @param string|array $args
      * @example $list = $list->sort('Name'); // default ASC sorting
-     * @example $list = $list->sort('Name DESC'); // DESC sorting
+     * @example $list = $list->sort('Name ASC, Age DESC');
      * @example $list = $list->sort('Name', 'ASC');
-     * @example $list = $list->sort(array('Name'=>'ASC', 'Age'=>'DESC'));
-     *
-     * @param string|array Escaped SQL statement. If passed as array, all keys and values are assumed to be escaped.
-     * @return static
+     * @example $list = $list->sort(['Name' => 'ASC', 'Age' => 'DESC']);
+     * @example $list = $list->sort('MyRelation.MyColumn ASC')
+     * @example $list->sort(null); // wipe any existing sort
      */
-    public function sort()
+    public function sort(...$args): static
     {
-        $count = func_num_args();
-
+        $count = count($args);
         if ($count == 0) {
             return $this;
         }
-
         if ($count > 2) {
             throw new InvalidArgumentException('This method takes zero, one or two arguments');
         }
-
         if ($count == 2) {
-            $col = null;
-            $dir = null;
-            list($col, $dir) = func_get_args();
-
-            // Validate direction
-            if (!in_array(strtolower($dir ?? ''), ['desc', 'asc'])) {
-                user_error('Second argument to sort must be either ASC or DESC');
+            list($column, $direction) = $args;
+            $sort = [$column => $direction];
+        } else {
+            $sort = $args[0];
+            if (!is_string($sort) && !is_array($sort) && !is_null($sort)) {
+                throw new InvalidArgumentException('sort() arguments must either be a string, an array, or null');
+            }
+            if (is_null($sort)) {
+                // Set an an empty array here to cause any existing sort on the DataLists to be wiped
+                // later on in this method
+                $sort = [];
+            } elseif (empty($sort)) {
+                throw new InvalidArgumentException('Invalid sort paramater');
```

Comment:
```suggestion
                throw new InvalidArgumentException('Invalid sort parameter');
```
Sorry, should have caught this last time around

Reply:
Updated